src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -0,0 +1,523 @@
+        {
+          transition_spectrum = exp.getSpectra()[0];;
+        }
+      }
+
+      // transition calculations
+      // calculate max intensity peak and threshold
+      float max_int = 0.0;
+      float min_int = numeric_limits<float>::max();
+      for (MSSpectrum::const_iterator spec_it = transition_spectrum.begin(); spec_it != transition_spectrum.end(); ++spec_it)
+      {
+        //find the max intensity peak
+        if (spec_it->getIntensity() > max_int)
+        {
+          max_int = spec_it->getIntensity();
+        }
+        if (spec_it->getIntensity() < min_int)
+        {
+          min_int = spec_it->getIntensity();
+        }
+      }
+
+      // no peaks or all peaks have same intensity (single peak / noise)
+      if (min_int >= max_int) { continue; }
+
+      // threshold should be at x % of the maximum intensity
+      // hard minimal threshold of min_int * 1.1
+      float threshold_transition = max_int * (transition_threshold/100);
+      float threshold_noise = min_int * 1.1;
+
+      AssayRow row;
+      int transition_counter = 1;
+
+      for (MSSpectrum::iterator spec_it = transition_spectrum.begin(); spec_it != transition_spectrum.end(); ++spec_it)
+      {
+        float current_int = spec_it->getIntensity();
+        double current_mz = spec_it->getMZ();
+
+        // write row for each transistion
+        // current int has to be higher than transition thresold and should not be smaller than threshold noise
+        if (current_int > threshold_transition && current_int > threshold_noise)
+        {
+          float rel_int = current_int/max_int;
+          row.precursor_mz = highest_precursor_mz;
+          row.product_mz = current_mz;
+          row.library_int = rel_int;
+          row.normalized_rt = spectrum_rt;
+          row.compound_name = description;
+          row.smiles = "none"; // not in AccurateMassSearch output yet
+          row.sumformula = sumformula;
+          row.adduct = adduct;
+          row.transition_group_id = String(transition_group_counter)+"_"+sumformula;
+          row.transition_id = String(transition_counter)+"_"+sumformula;
+          row.decoy = 0;
+          transition_counter += 1;
+        }
+        else
+        {
+          continue;
+        }
+      assaylib.push_back(row);
+      }
+       transition_group_counter += 1;
+    }
+
+    // write output
+    TextFile tf;
+    tf.addLine("PrecursorMz\tProductMz\tLibraryIntensity\tRetentionTime\tCompoundName\tSMILES\tSumFormula\tTransitionId\tTransitionGroupId\tAdduct\tDecoy\n");
+    for (auto & entry : assaylib)
+    {
+      tf.addLine(String(entry.precursor_mz)+"\t"+String(entry.product_mz)+"\t"+String(entry.library_int)+"\t"+String(entry.normalized_rt)+
+                 "\t"+entry.compound_name+"\t"+entry.smiles+"\t"+entry.sumformula+"\t"+entry.transition_group_id+"\t"+entry.transition_id+
+                 "\t"+entry.adduct+ "\t"+entry.decoy+"\n");
+    }
+    tf.store(out);
```

Comment:
this works for now, in the future it would be nice to use the TransitionTSVWriter ...